ocs/ocs_agent.py

```diff
@@ -28,9 +31,20 @@
 from ocs import client_t
 from ocs import ocs_feed
 from ocs.base import OpCode
+from typing import Tuple, Optional, Callable, Dict, Any, Union, TypeVar, Generator
+
+
+OpFuncType = Union[
+    Callable[["OpSession", Optional[Dict[str, Any]]], Tuple[bool, str]],
+    Callable[["OpSession", Optional[Dict[str, Any]]], Deferred[Tuple[bool, str]]],
+]
```

Comment:
Yep, that's right.

OpSession being a string here is an example of a [forward reference](https://peps.python.org/pep-0484/#forward-references) which is necessary because the class has not been declared yet in the file.

I don't think this is needed in newer versions of python, however in the older python tests without this.